src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String BUY = "b";
+    private static final String SUPPLY = "s";
+    private static final String WHITE_SPACE = " ";
+
+    protected static int[] readFromFile(String fromFileName) {
+        File file = new File(fromFileName);
+        int countBuy = 0;
+        int countSupply = 0;
+        String line = null;
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] words = line.split(WHITE_SPACE);
+                for (String word : words) {
+                    if (word.startsWith(BUY)) {
+                        countBuy += Integer.parseInt(word.substring(4));
+                    } else if (word.startsWith(SUPPLY)) {
+                        countSupply += Integer.parseInt(word.substring(7));
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read file " + fromFileName, e);
+        }
+        return new int[]{countBuy, countSupply};
+    }
 
-    public void getStatistic(String fromFileName, String toFileName) {
+    protected static String createReport(String fromFileName) {
+        int[] valueBuySupply = readFromFile(fromFileName);
+        valueBuySupply[0] /= 2;
+        valueBuySupply[1] /= 2;
```

Comment:
why /= 2?

Reply:
because there is sum of all "buy" and "supply" in the file, isn't it? 2nd and 3rd line from the end. If I sum all buy and all supply the result (1st line from the end) will be different

Reply:
there should be no result in files "apple", "banana" etc. result should be in, for example, "appleResult". maybe you had incorrect logic, most certainly incorrect `toFile` usage.

Reply:
done